src/index.js

```diff
@@ -5,11 +5,8 @@ import { resolve } from 'import-meta-resolve'
 import path from 'path'
 import puppeteer from 'puppeteer'
 import url from 'url'
+import pkg from '../package.json' with { type: 'json' }
```

Comment:
Oh no! I'm really sorry, but it looks like `@mermaid-js/mermaid-cli` supports Node.JS v18.19, but only Node.JS v18.20 added support for `with { type: 'json' }`. Previously, it was `assert { type: 'json' }`!

See https://nodejs.org/en/blog/release/v18.20.0

Unfortunately, since dropping support for Node.JS v18.19 would be a breaking change, we won't be able to merge this PR until we are ready to release `@mermaid-js/mermaid-cli` v12, which might be a while!

---

Maybe as an alternative to fix #796, could we instead just avoid loading `package.json`?

Maybe we can make a new file like `src/version.js` that just consists of:

```node
export const version = '11.4.0';
```

And we can make a [`npm version`](https://docs.npmjs.com/cli/v8/commands/npm-version) script that just automatically creates this file and adds it to git using `git add`, to make sure it's kept in sync with the `package.json` file.

Reply:
Oh...
No, I should be the one apologizing.
I should have verified more thoroughly about the Node.js feature release.

> Maybe as an alternative to fix https://github.com/mermaid-js/mermaid-cli/issues/796, could we instead just avoid loading package.json?
> Maybe we can make a new file like src/version.js that just consists of:

That's cool!!
You're absolutely right!
I agree with that idea as well!
This task will discard all the current PR's changes, so I will create a separate PR for this work!

When releasing, the package version is referenced as `env.RELEASE_VERSION` within the github actions, so I plan to try using this variable.
It seems that it can also be referenced using package.json vars, but I believe it's better to refer to the value from the github actions.
I think that accessing it from two different contexts could potentially lead to bugs in the future.

See: https://github.com/mermaid-js/mermaid-cli/blob/master/.github/workflows/release-publish.yml
See: https://docs.npmjs.com/cli/v9/using-npm/scripts#packagejson-vars

Reply:
I replaced PR #798 .
Once you confirm my comment, I will close this PR.

Reply:
:rocket:

> I should have verified more thoroughly about the Node.js feature release.

Nah, it's my fault too! I even looked at the https://nodejs.org/docs/latest-v18.x/api/esm.html#import-specifiers page and it seemed fine! But I should have also looked at the https://nodejs.org/docs/v18.19.0/api/esm.html#import-specifiers page too :sweat_smile:

> This task will discard all the current PR's changes, so I will create a separate PR for this work!

Sounds perfect to me! Your PR https://github.com/mermaid-js/mermaid-cli/pull/798 looks great to me!

I have spotted a bug in it, but I've got a potential fix in https://github.com/mermaid-js/mermaid-cli/pull/798#discussion_r1864737878 that should solve it!

Reply:
Thank you for everything!
I also checked comments.
Let's close this PR and continue in #798 .